core/src/test/java/org/elasticsearch/rest/action/support/RestTableTests.java

```diff
@@ -70,6 +71,55 @@ public void testThatDisplayHeadersAreNotAddedTwice() throws Exception {
         assertThat(headerNames, not(hasItem("unmatched")));
     }
 
+    public void testThatWeUseTheAcceptHeaderJson() throws Exception {
+        assertBuildResponse(Collections.singletonMap("Accept", "application/json"), "application/json",
+                "[{\"bulk.foo\":\"foo\",\"bulk.bar\":\"foo\",\"aliasedBulk\":\"foo\"," +
+                        "\"aliasedSecondBulk\":\"foo\",\"unmatched\":\"foo\"," +
+                        "\"invalidAliasesBulk\":\"foo\"}]");
+    }
+
+    public void testThatWeUseTheAcceptHeaderText() throws Exception {
+        assertBuildResponse(Collections.singletonMap("Accept", "text/plain"), "text/plain", "foo foo foo foo foo foo \n");
+    }
+
+    public void testThatWeUseTheContentTypeHeaderJson() throws Exception {
+        assertBuildResponse(Collections.singletonMap("Content-Type", "application/json"), "application/json",
```

Comment:
Well right now we cover all cases, my next commit changes string literals to constants, hoping to reduce duplication and any future errors that way. 

If I understand you correctly, rarely would execute one of the content types more often that the other, could you expand on the rationale a little more - I'm not sure I understand the benefit.